src/Symfony/Component/Form/ChoiceList/ArrayChoiceList.php

```diff
@@ -233,13 +233,13 @@ private function castableToString(array $choices, array &$cache = array())
                 }
 
                 continue;
-            } elseif (!is_scalar($choice)) {
+            } elseif (!is_scalar($choice) && null !== $choice) {
```

Comment:
Actually, I really think there is a good reason to not reverse this change.

This `placeholder` issue is not related to this PR, in 2.7+, if one of the choice has a value `null`, and `null` is the default data, `placeholder` is never selected by default in favour of this choice.

Plus, if you submit the form with the `placeholder` selected and even if `required` is `true` the form is valid and fallback on the `null` data when synchronising which may be a kind of bug as it would not be expected.

The PR fixes it as `null` is always converted to empty string so the `placeholder` is not added in the choice list as `placeholder_in_choices` is `true`.

To avoid the conflict, setting `placeholder` with an internal default string value like `~placeholder~` might be considered after all...